### DISCUSSION TITLE
Does ImageMagick recognize Facesaver files?

### DISCUSSION THREAD

**Discussion by grybo1960** [ImageMagick/ImageMagick#2138] [General]
Does ImageMagick recognize Facesaver files?  Facesaver files were used at Usenix conventions strting in the late 1980's.   I don't know what the file extention would be.

**Comment by fmw42** [comment]
I do not think so. But from http://fileformats.archiveteam.org/wiki/FaceSaver, it looks like the NetPBM tools have a converted fstopgm to PGM format which ImageMagick can read. NetPBM tools are generally available.  See http://netpbm.sourceforge.net and https://www.commandlinux.com/man-page/man1/netpbm.1.html

**Comment by snibgo** [comment]
Following Fred's post, I discover my Windows laptop already has fstopgm. I download a file from the Wiki link.

```
fstopgm jimf.face |magick - out.png
```
This makes a readable out.png file. However, it says:
```
fstopgm: warning, non-square pixels
```
Some more digging is needed to find how non-square the pixels are. Taking a guess from the text metadata, we get:

```
fstopgm jimf.face |%IMG7%magick - -resize "96x128^!" out.png
```

![image](https://user-images.githubusercontent.com/33812368/84555799-c12c1980-ad16-11ea-9043-593e89d816e3.png)

**Reply by fmw42** [reply]
Thanks, snibgo. I tried to do the same on my Mac, but my version of NetPBM did not have pftopgm. I was looking into downloading an updated version. But you have proved my point. I believe that most OS systems come with NetPBM.
